### PR TITLE
don't use the database during download

### PR DESCRIPTION
**Pull request by harry-73**
The use of the database during the download process means that the display of the map in real time may be altered. 
The goal in this case is to download the images.
At the end of the process, the database is used again

### CODE REVIEW

**Comment by harry-73** [comment]
Hi, did you have time to review it?

**Comment by allartk** [comment]
No, sorry, not yet

**Comment by allartk** [comment]
There are some linting errors, maybe you can add eslint to your editor, such as [vscode](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) please, or run manual `npm run lint` to check your syntax before commit? 
Besides, I have some doubts, why throw an exception, if you want to prevent the map view from being changed, I think you should do it at the map level, and we could document that?

**Comment by harry-73** [comment]
> There are some linting errors, maybe you can add eslint to your editor, such as [vscode](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) please, or run manual `npm run lint` to check your syntax before commit? 

I have installed it yesterday for my own development. Now, It should be OK. The last commit is OK. Sorry for that.

> Besides, I have some doubts, why throw an exception, if you want to prevent the map view from being changed, I think you should do it at the map level, and we could document that?

I am not sure to understand, I do want to prevent the map view from being changed but I want to be able, during the downloaded process, to have the tiles displayed when I move into it

**Comment by allartk** [comment]
Let me ask it differently, you are trying to block the map from moving or zooming (and downloading new tiles) during download? If yes, I see it as a user interface thing (eg block map input with a html overlay)

**Comment by harry-73** [comment]
If that was my intention, I agree, it should be done at the Map level.
In fact, I noticed that during the tile download process, writing to the database was competing with reading from the tiles to display the map when moving it.
The purpose here is to avoid using the database while the download process is running. This allows to continue to have a smooth user experience.

**Comment by allartk** [comment]
It's clear now, i forgot about the catch for this exception. I will leave some small comments and it can be merged.

File: src/TileLayerOffline.js [hunk 1]
```diff
@@ -60,6 +74,11 @@ const TileLayerOffline = L.TileLayer.extend(
      * @return {Promise<string>} objecturl
      */
     setDataUrl(coords) {
+      if (this._ongoingSave)
+        return Promise.resolve().then(() => {
+          throw new Error('On download state');
```
Comment (allartk) [inline src/TileLayerOffline.js:79]: Without testing, I excpect this can be replaced with
```
return Promise.reject('On download state');
```

**Review by allartk — CHANGES_REQUESTED** [review]
Sorry for being so slow and misunderstanding :) Just a small change and after it I'll merge (and make a 2.x release when I've got some time).

Are you willing to do some merging (in a separate branch) with the typescripted dev branch?

**Review by allartk — APPROVED** [review]

**Comment by harry-73** [comment]
> Are you willing to do some merging (in a separate branch) with the typescripted dev branch?

I know nothing about typescript  or very few...

**Pull request merged**